### DISCUSSION TITLE
Xcode 15 will be default starting October 16

### DISCUSSION THREAD

**Discussion by helinanever** [codemagic-ci-cd/codemagic-docs#2084] [Announcements]
Now that we’ve had the Xcode 15 Release available on our M1 and M2 machines for some time, we will make the Xcode 15 image our default one for Apple Silicon-based machines and switch Xcode `latest` to that version on **October 16**.

This means that your builds will be run on the machine utilizing the Xcode 15 image if you have selected `15`, `latest`, or `edge` as the Xcode version, or haven’t specified an Xcode version.

The Xcode 15 image comes with new runtimes for Apple Watch Series 9, Apple Watch Ultra 2, and iPhone 15 devices. See the [build machine specs](https://docs.codemagic.io/specs/versions-macos/) for information about the preinstalled software and available runtimes.

**If you are not ready to switch to Xcode 15 yet, please review your build configuration in time and fix the Xcode version setting to a specific version.** You can configure the Xcode version in the Build section of the Flutter workflow editor or in the environment section of codemagic.yaml.

We’d be happy to help with any questions or concerns!

**Comment by ollie-at-xdesign** [comment]
Hi @helinanever,

Is there any plans for CodeMagic to make Xcode 15.0.1 available?
This version introduces a few really important bug fixes.

Cheers
Ollie

**Reply by helinanever** [reply]
Hi @ollie-at-xdesign, we'll make this version available as soon as this becomes GA.

**Reply by ollie-at-xdesign** [reply]
Hi @helinanever - could you confirm the build number that is currently used on CM for XC 15? 
XC 15.0.1 RC1 was released to the public on the 11th October.

Cheers

**Reply by helinanever** [reply]
We're currently still at 15.0 (15A240d), see the docs [here](https://docs.codemagic.io/specs/versions-macos/). We'll post an update as soon as 15.0.1 is available.

**Comment by pdax-king** [comment]
We experienced an issue where the upload iOS Certificates, iOS Provisioning Profile and Android KeyStore is missing. We don't have problem with our builds last week. Does this upgrade to XCode 15 might be the possible cause of our issue?

The issue can easily be resolve by re-uploading them but I want to know if the upgrade to XCode 15 was the cause or there might be other reason/bugs that cause this experience.

**Reply by helinanever** [reply]
Hi @pdax-king, the issue you are describing is not related to Xcode version updates. Please start a new discussion and someone from our support team will get in touch with you. Or if you're on a paid plan, you can use the in-app chat widget to contact support.

**Reply by pdax-king** [reply]
Thanks for confirming @helinanever. I will create an issue if we encounter it again.

**Reply by Maxim-Filimonov** [reply]
@helinanever I'm experienced the same issues with code signing **ONLY** when I update machines to use xcode 15. I downgraded to 14.3 and build passes without problems the code signing step.

**Reply by helinanever** [reply]
@Maxim-Filimonov if the issue persists for you, I recommend starting a new discussion so the team can follow up on it.

**Comment by helinanever** [comment]
The Xcode 15 image is now default and `latest` in Xcode version settings points to 15.

**Comment by georgidimitrovx** [comment]
If anyone is having this error after switching to Xcode 15 `Error (Xcode): Cycle inside Runner; building could produce unreliable results.` then change the order of your `buildPhases` inside `project.pbxproj`. 

Here's my case (I obfuscated the codes to FFFF):

Before
```
buildPhases = (
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* [CP] Check Pods Manifest.lock */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* Run Script */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* Sources */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* Frameworks */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* Resources */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* Embed Frameworks */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* Thin Binary */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* [CP] Embed Pods Frameworks */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* [CP] Copy Pods Resources */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* Embed App Extensions */,
);
```

After
```
buildPhases = (
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* [CP] Check Pods Manifest.lock */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* Run Script */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* Sources */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* Frameworks */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* Resources */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* Embed Frameworks */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* Embed App Extensions */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* Thin Binary */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* [CP] Embed Pods Frameworks */,
    FFFFFFFFFFFFFFFFFFFFFFFFFFF /* [CP] Copy Pods Resources */,
);
```

I moved `Embed App Extensions` above `Thin Binary` and `[CP] Embed Pods Frameworks` because of cycle depencency.

**Comment by eli1stark** [comment]
@helinanever  The new Xcode 15 version is causing issues with all my builds; it cuts off half of the source code during the build process. Prior to October 16, my IPA file size was 130MB, but now it's 80MB. The app is crashing on the launch when built with Codemagic on the new Xcode version.

**Reply by eli1stark** [reply]
To fix the build issues, set your Xcode version to 14.3.1. It took me 4 hours today to realize that the issue with my app is because of the Codemagic update. 🥲🥲

**Reply by helinanever** [reply]
Hi @eli1stark, sorry to hear you had issues after we made Xcode 15 the default version. To avoid such issues in the future, we recommend explicitly setting the Xcode version you want to use in your build configuration. Leaving it unset or using `latest` can result in surprises like this. :)

**Comment by mohamed-gearbox** [comment]
Since release of XCode 15 as the latest (default) version on codemagic, I realized that after the build completes successfully and is published to Apple TestFlight. Everytime we try to launch the app on iOS device it crashes right away. Here is a copy of crash report exception

Incident Identifier: 7684AB3D-FB02-48CA-B1E3-2B013599AA92
Hardware Model:      iPhone9,4
Process:             Runner [475]
Path:                /private/var/containers/Bundle/Application/8FBA207F-4A47-4CB7-BD20-FCA532BC7493/Runner.app/Runner
Identifier:          xxx.xxxxxxxx.ios
Version:             1.0.0 (17)
AppStoreTools:       15A240a
AppVariant:          1:iPhone9,4:15
Beta:                YES
Code Type:           ARM-64 (Native)
Role:                Foreground
Parent Process:      launchd [1]
Coalition:           xxx.xxxxxxxx.ios [508]

Date/Time:           2023-10-17 21:36:18.7595 -0300
Launch Time:         2023-10-17 21:36:18.5557 -0300
OS Version:          iPhone OS 15.7.9 (19H365)
Release Type:        User
Baseband Version:    6.03.01
Report Version:      104

Exception Type:  EXC_BAD_ACCESS (SIGSEGV)
Exception Subtype: KERN_INVALID_ADDRESS at 0x0000000000000000
Exception Codes: 0x0000000000000001, 0x0000000000000000
VM Region Info: 0 is not in any region.  Bytes before following region: 4340187136
      REGION TYPE                 START - END      [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      __TEXT                   102b20000-102b28000 [   32K] r-x/r-x SM=COW  ...er.app/Runner
Exception Note:  EXC_CORPSE_NOTIFY
Termination Reason: SIGNAL 11 Segmentation fault: 11
Terminating Process: exc handler [475]

Triggered by Thread:  0


Thread 0 name:
Thread 0 Crashed:
0   ???                           	0x0000000000000000 0x0 + 0
...

Resolution for the time being was to set xcode: 14.3.1 (was set to latest before) in codemagic yaml file, until we figure out how we can resolve this issue when compiling using XCode 15 during codemagic build process.

Took sometime to realize that the issue was due to cutting over to xcode version 15 during the build process with us using latest identifier for xcode version. Thanks @eli1stark for the tip! :)

**Comment by elmariocarlos** [comment]
Hello all & @helinanever.

We have been getting errors in the publishing phase since some builds back. We thought it might had to do with some Xcode settings; but after careful reading of the logs, it seems the FLUTTER_BUILD_NUMBER & FLUTTER_BUILD_NAME variables are not being propertly set in the Xcode exports. The IPA file reflects the correct incremental versions, but the variables end up with "bad strings" that are not compatible with the upload/publishing tool.

After reaching this thread and looking at other issues; we tried out going back to 14.3.1. With this, the publishing stage completes without an issue (Build ID : 65322b22eef458a9573b1ccf), so it definetely is related to the Xcode 15 update. Here are the error messages when using Xcode 15 to build (Build ID: 653222d951c4e373ca04b386):

```
Upload "/Users/builder/clone/build/ios/ipa/Gamership.ipa" to App Store Connect
2023-10-20 07:06:15.021 *** Error: ERROR: [ContentDelivery.Uploader] Asset validation failed (90060) This bundle is invalid. The value for key CFBundleShortVersionString '$(FLUTTER_BUILD_NAME)' in the Info.plist file at 'Payload/Gamership.app/Frameworks/App.framework' must be a period-separated list of at most three non-negative integers. Please find more information about CFBundleShortVersionString at https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring (ID: f0e08994-923b-4ef1-9a40-ef4276950327)
2023-10-20 07:06:15.023 *** Error: ERROR: [ContentDelivery.Uploader] Asset validation failed (90058) This bundle is invalid. The value for key CFBundleVersion [$(FLUTTER_BUILD_NUMBER)] in the Info.plist file must be a period-separated list of at most three non-negative integers. Please find more information about CFBundleVersion at https://developer.apple.com/documentation/bundlereso
/urces/information_property_list/cfbundleversion (ID: 4a4eb264-c8e2-4e65-a105-5c2ba7c44b35)
2023-10-20 07:06:15.024 *** Error: ERROR: [ContentDelivery.Uploader] Asset validation failed (90059) This bundle Payload/Gamership.app/Frameworks/App.framework is invalid. The value for key CFBundleVersion [$(FLUTTER_BUILD_NUMBER)] in the Info.plist file must be no longer than 18 characters. Please find more information about CFBundleVersion at https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion (ID: 3695cd36-a476-462b-bc05-dae75357eeba)
2023-10-20 07:06:15.238  INFO: [ContentDelivery.Uploader] 
=============
UPLOAD FAILED with 3 errors.
=============
```

**Reply by helinanever** [reply]
Hi @elmariocarlos, sadly it's unclear what might be causing the issue you're describing. There are a few things you can try to debug this. First, you can run the iOS build with the `--verbose` flag to get more logs. You can also SSH into the build machine to check if  `Generated.xcconfig` is created somewhere in the build/ folder as this is the file that defines those values.

**Comment by helinanever** [comment]
The Xcode 15 image is now at version 15.0.1 (15A507) which will be also used if you specify `15`, `latest` or `edge` in Xcode version settings. This fixes an issue related to the iOS simulator taking a long time to launch, see the release notes [here](https://developer.apple.com/documentation/xcode-release-notes/xcode-15_0_1-release-notes).
